settlement/dymension/dymension.go

```diff
@@ -249,9 +249,8 @@ func (d *HubClient) eventHandler() {
 	for {
 		select {
 		case <-d.ctx.Done():
-			return
 		case <-d.client.EventListenerQuit():
-			panic("Settlement WS disconnected")
+			return
```

Comment:
not sure I understand the logic of `return` here. 
 
as I see it if we `return` here vs panicing, we don't have a websocket connection to the hub anymore, hence we won't get events of update hence we'll never move to submitting the next batch. so basically the rollapp will continue creating batches without submitting them.

Panic in that case makes the rollapp restart and retry the connection to the hub.

Either way don't think panic is the long term solution here, but I'm actually working on a reconnect solution.

Reply:
The issue as I see it, is that for graceful shutdown using SIGTERM, no panic should happen. 
But U totally right. It should panic when websocket with the hub is disconnected.
I Will fix the PR to support both requirements.

Regarding the panic through the code,
Panic can be recovered in the same `goroutine` so it's hard to have a single `recover` func that covers all the panics.
IMO, panic in the code should be **only** in unrecoverable logic flow.
We can have a separate issue to review all the panics currently in-place

Reply:
true. we should handle graceful shutdown using SIGTERM. With that being said I believe we should also handle panic with the same urgency is specific places.  

> Panic can be recovered in the same goroutine so it's hard to have a single recover func that covers all the panics.

Not sure if that's what you meant but panic bubbles up so basically can be caught in the upper most function (i.e node.go). from there the node can wait for all of it's children goroutines to gracefully shutdown - or at least the critical ones.

The reason I think it's important is because we had a few situations where the rollapp kept panicing and some of the panics lead to an inconsistent unrecoverable state in the rollapp which required manual intervention. 

I believe the critical goroutines to make sure are gracefully shutdown are in the `manager`.  so IMO we can start by at least making sure this gracefully shuts down upon panic/SIGTERM.